frontend/src/modules/Metadata_Forms/components/metadataAttachement.js

```diff
@@ -0,0 +1,304 @@
+import React, { useEffect, useState } from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import {
+  Autocomplete,
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Divider,
+  Grid,
+  TextField,
+  Typography
+} from '@mui/material';
+import {
+  deleteAttachedMetadataForm,
+  getAttachedMetadataForm,
+  getMetadataForm,
+  listAttachedMetadataForms,
+  listMetadataForms
+} from '../services';
+import { Defaults, PlusIcon } from '../../../design';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useClient } from '../../../services';
+import { RenderedMetadataForm } from './renderedMetadataForm';
+import { SET_ERROR } from '../../../globalErrors';
+import { AttachedFormCard } from './AttachedFormCard';
+import DoNotDisturbAltOutlinedIcon from '@mui/icons-material/DoNotDisturbAltOutlined';
+import DeleteIcon from '@mui/icons-material/DeleteOutlined';
+
+export const MetadataAttachement = (props) => {
+  const { entityType, entityUri, canEdit } = props;
+  const client = useClient();
+  const dispatch = useDispatch();
+  const [selectedForm, setSelectedForm] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [loadingFields, setLoadingFields] = useState(false);
+  const [formsList, setFormsList] = useState([]);
+  const [fields, setFields] = useState([]);
+  const [filter] = useState({
+    ...Defaults.filter,
+    entityType: entityType,
+    entityUri: entityUri
+  });
+  const [addNewForm, setAddNewForm] = useState(false);
+  const [availableForms, setAvailableForms] = useState([]);
+
+  const fetchAvailableForms = async () => {
+    const response = await client.query(listMetadataForms({}));
+    if (!response.errors) {
+      setAvailableForms(
+        response.data.listMetadataForms.nodes.map((form) => ({
+          label: form.name,
+          value: form.uri,
+          form: form
+        }))
```

Comment:
should available forms only be forms that a user can see that are also not already attachd to the entity - compliment of the set?

Reply:
listMetadataForms has a visibility filter by default there.

Reply:
Implemented 'see that are also not already attachd to the entity '